src/Sylius/Bundle/CoreBundle/Fixture/Factory/CatalogPromotionActionExampleFactory.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Fixture\Factory;
+
+use Sylius\Component\Promotion\Model\CatalogPromotionActionInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class CatalogPromotionActionExampleFactory extends AbstractExampleFactory implements ExampleFactoryInterface
+{
+    private FactoryInterface $catalogPromotionActionFactory;
+
+    private OptionsResolver $optionsResolver;
+
+    public function __construct(FactoryInterface $catalogPromotionActionFactory)
+    {
+        $this->catalogPromotionActionFactory = $catalogPromotionActionFactory;
+
+        $this->optionsResolver = new OptionsResolver();
+
+        $this->configureOptions($this->optionsResolver);
+    }
+
+    public function create(array $options = []): CatalogPromotionActionInterface
+    {
+        $options = $this->optionsResolver->resolve($options);
+
+        /** @var CatalogPromotionActionInterface $catalogPromotionAction */
+        $catalogPromotionAction = $this->catalogPromotionActionFactory->createNew();
+        $catalogPromotionAction->setType($options['type']);
+        $catalogPromotionAction->setConfiguration($options['configuration']);
+
+        return $catalogPromotionAction;
+    }
+
+    protected function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setDefault('type', CatalogPromotionActionInterface::TYPE_PERCENTAGE_DISCOUNT)
+            ->setAllowedTypes('type', 'string')
+            ->setDefault('configuration', [])
```

Comment:
I would be for setting some amount as a default configuration because now, the default action in invalid